lib/kubernetes-deploy/restart_task.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubeclient_builder'
+
+module KubernetesDeploy
+  class RestartTask
+    include KubernetesDeploy::KubeclientBuilder
+
+    HTTP_OK_RANGE = 200..299
+
+    def initialize(context:, namespace:, logger:)
+      @context = context
+      @namespace = namespace
+      @logger = logger
+      @v1beta1_kubeclient = build_v1beta1_kubeclient(context)
+    end
+
+    def perform(deployments)
+      if deployments.empty?
+        raise ArgumentError, "#perform takes at least one deployment to restart"
+      end
+
+      fetch_deployments(deployments).each do |record|
```

Comment:
If it's hard to build, it's probably not worth it—but most devs would have no idea to look there and probably won't even have access. It would feel like you should be able to use the same code as we do for deploys to do this without too much trouble, but I'm not sure. I'll leave this up to you.